src/main/java/shortestpath/FairyRingUtil.java

```diff
@@ -0,0 +1,83 @@
+package shortestpath;
+
+import net.runelite.api.Client;
+import net.runelite.api.ScriptID;
+import net.runelite.api.gameval.InterfaceID;
+import net.runelite.api.widgets.Widget;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class FairyRingUtil {
+
+    /**
+     * Find the applicable fairy ring log panels then scroll to and highlight the fairy ring destination in our path.
+     */
+    public static void scrollFairyRingPanel(ShortestPathPlugin plugin, Client client) {
+        final String fairyRingCode = getFairyRingCodeFromPath(plugin);
+        if (fairyRingCode == null) return;
+
+        final Widget contentsList = client.getWidget(InterfaceID.FairyringsLog.CONTENTS);
+        if (contentsList == null) return;
+
+        final Widget favesPanel = client.getWidget(InterfaceID.FairyringsLog.FAVES);
+        findMatchingWidget(favesPanel, contentsList, fairyRingCode).ifPresent(foundCodeWidget -> {
+            final int panelScrollY = Math.min(
+                    foundCodeWidget.getRelativeY(),
+                    contentsList.getScrollHeight() - contentsList.getHeight()
+            );
+
+            contentsList.setScrollY(panelScrollY);
+            contentsList.revalidateScroll();
+
+            foundCodeWidget.setTextColor(0x00FF00);
+            foundCodeWidget.setText("(Shortest Path) " + foundCodeWidget.getText());
+
+            client.runScript(
+                    ScriptID.UPDATE_SCROLLBAR,
+                    InterfaceID.FairyringsLog.SCROLLBAR,
+                    InterfaceID.FairyringsLog.CONTENTS,
+                    panelScrollY
+            );
+        });
+    }
+
+    /**
+     * Find the applicable fairy ring transport based on our current pathfinder path.
+     */
+    private static String getFairyRingCodeFromPath(ShortestPathPlugin plugin) {

```

Comment:
This will not be restricted to only fairy ring codes, so it can return any other `transport.getDisplayInfo()` such as e.g. `Varrock Teleport`. You should include a check like `TransportType.FAIRY_RING.equals(transport.getType())`